exporters/geneva/include/opentelemetry/exporters/geneva/metrics/exporter.h

```diff
@@ -0,0 +1,140 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "opentelemetry/common/spin_lock_mutex.h"
+#include "opentelemetry/exporters/geneva/metrics/connection_string_parser.h"
+#include "opentelemetry/exporters/geneva/metrics/data_transport.h"
+#include "opentelemetry/exporters/geneva/metrics/exporter_options.h"
+#include "opentelemetry/sdk/metrics/metric_exporter.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter {
+namespace geneva {
+namespace metrics {
+
+constexpr size_t kBufferSize = 65360; // the maximum ETW payload (inclusive)
+constexpr size_t kMaxDimensionNameSize = 256;
+constexpr size_t kMaxDimensionValueSize = 1024;
+constexpr size_t kBinaryHeaderSize = 4; // event_id (2) + body_length (2)
+constexpr size_t kMetricPayloadSize =
+    24; // count_dimension (2)  + reserverd_word (2) + reserverd_dword(4) +
+        // timestamp_utc (8) + metric_data (8)
+constexpr size_t kExternalPayloadSize =
+    40; // count_dimension (2) + reserverd_word (2) + count (4) + timestamp_utc
+        // (8) + metric_data_sum (8) + metric_data_min(8) + metric_data_max(8)
+
+// time conversion constants
+constexpr uint32_t kWindowsTicksPerSecond =
+    10000000; // windows ticks are in 100 ns
+constexpr uint64_t kSecondsToUnixTime =
+    11644473600L; // number of seconds between windows epoch start
+                  // 1601-01-01T00:00:00Z and UNIX/Linux epoch
+                  // (1970-01-01T00:00:00Z)
+
+enum class MetricsEventType : uint16_t {
+  ULongMetric = 50,
+  DoubleMetric = 55,
+  ExternallyAggregatedULongDistributionMetric = 56
+};
+
+/**
+ * The Geneva metrics exporter exports metrics data to Geneva
+ */
+class Exporter final : public opentelemetry::sdk::metrics::MetricExporter {
```

Comment:
nit: change the generic `Exporter` to `GenevaExporter`?

Reply:
yeah I thought so, but it seems redundant as namespace already contains `Geneva`.